specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1097 @@
+    by the API stack (e.g. DNS record configuration in certain environments).
+    Implementations are expected to treat these as "MUST" for factors within the
+    control of the implementation.
+
+1.  For non-`Ready` conditions, any conditions with `severity=""` (aka "Error
+    conditions") must be aggregated into the "Ready" condition as follows:
+
+    1.  If the condition is `False`, `Ready` MUST be `False`.
+    1.  If the condition is `Unknown`, `Ready` MUST be `False` or `Unknown`.
+    1.  If the condition is `True`, `Ready` may be any of `True`, `False`, or
+        `Unknown`.
+
+    Implementations MAY choose to report that `Ready` is `False` or `Unknown`
+    even if all Error conditions report a status of `True` (i.e. there may be
+    additional hidden implementation conditions which feed into the `Ready`
+    condition which are not reported.)
+
+1.  Non-`Ready` conditions with non-error severity MAY be surfaced by the
+    implementation. Examples of `Warning` or `Info` conditions could include:
+    missing health check definitions, scale-to-zero status, or non-fatal
+    capacity limits.
+
+Conditions type names should be chosen to describe positive conditions where
+`True` means that the condition has been satisfied. Some conditions may be
+transient (for example, `ResourcesAllocated` might change between `True` and
+`False` as an application scales to and from zero). It is RECOMMENDED that
+transient conditions be indicated with a `severity="Info"`.
+
+# Resource Lifecycle
+
+## Broker Lifecycle
+
+A Broker represents an Addressable endpoint (i.e. it has a `status.address.url`
+field) which can receive, store, and forward events to multiple recipients based
+on a set of attribute filters (Triggers). Triggers are associated with a Broker
+based on the `spec.broker` field on the Trigger; it is expected that the
+controller for a Broker will also control the associated Triggers. When the
+Broker's `Ready` condition is `true`, the Broker MUST provide a
+`status.address.url` which accepts all CloudEvents and MUST forward the received
+events to each associated Trigger whose `Ready` condition is `true`. As
+described in the [Trigger Lifecycle](#trigger-lifecycle) section, a Broker MAY
+forward events to an associated Trigger which which does not currently have a
+`true` `Ready` condition, including events received by the Broker before the
+Trigger was created.
+
+When a Broker is created, its `spec.class` field MUST be populated to indicate
+which of several possible Broker implementations to use. It is RECOMMENDED to
+default the `spec.class` field on creation if it is unpopulated. Once created,
+the `spec.class` field MUST be immutable; the Broker must be deleted and
+re-created to change the `spec.class`. This pattern is chosen to make it clear
+that changing `spec.class` is not an atomic operation and that any
+implementation would be likely to result in message loss during the transition.
```

Comment:
just to be consistent, wonder if we should change 'message loss' -> 'event loss'?